### PR TITLE
feat(xc-admin-frontend): add basic instructions summary + refactor the code

### PR DESCRIPTION
**Pull request by keyvankhademi**
This add a simple summary section in the instructions showing how many of each instruction are in a proposal.

image

Also improves the ui of the proposal row
image

### CODE REVIEW

File: governance/xc_admin/packages/xc_admin_frontend/.eslintignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules
+tailwind.config.js
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/.eslintignore:2]: Just curious about what made you add these here, seems like they're already ignored by default?

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            const parsedInstruction = multisigParser.parseInstruction({
+              programId: innerInstruction.programId,
+              data: innerInstruction.data as Buffer,
+              keys: innerInstruction.keys as AccountMeta[],
+            })
+            acc[parsedInstruction.name] = (acc[parsedInstruction.name] ?? 0) + 1
+          })
+        } else if (governanceAction instanceof PythGovernanceActionImpl) {
+          acc[governanceAction.action] = (acc[governanceAction.action] ?? 0) + 1
+        } else if (governanceAction instanceof SetDataSources) {
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts:88]: I think `SetDataSources` inherits from `PythGovernanceActionImpl` so you don't need this

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          acc[governanceAction.actionName] =
+            (acc[governanceAction.actionName] ?? 0) + 1
+        } else {
+          acc['unknown'] = (acc['unknown'] ?? 0) + 1
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts:92]: something along the lines of unrecongnizedWormholeMessage ?

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * const sortedObj = sortObjectByValues(obj);
+ * console.log(sortedObj); // Outputs: { b: 3, a: 2, c: 1 }
+ */
+const sortObjectByValues = (obj: Record<string, number>) => {
+  const sortedEntries = Object.entries(obj).sort(([, a], [, b]) => b - a)
+  const sortedObj: Record<string, number> = {}
+  for (const [key, value] of sortedEntries) {
+    sortedObj[key] = value
+  }
+  return sortedObj
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts:55]: how confident are you that the of insertion is the order of retrieval? Quick google says that maybe`Map` is better for this

File: governance/xc_admin/packages/xc_admin_frontend/.eslintignore (hunk 1 above)
Comment (keyvankhademi) [reply governance/xc_admin/packages/xc_admin_frontend/.eslintignore:2]: When I ran the eslint command from inside the project folder it included these files. It's good to have them here to.

File: governance/xc_admin/packages/xc_admin_frontend/tsconfig.json [hunk 5]
```diff
@@ -13,7 +13,11 @@
     "isolatedModules": true,
     "jsx": "preserve",
     "incremental": true,
-    "skipLibCheck": true
+    "skipLibCheck": true,
+    "paths": {
+      "@images/*": ["./images/*"],
+      "xc-admin-common": ["../xc_admin_common/src"]
```
[the remark this answers is not in the document]
Comment (keyvankhademi) [reply governance/xc_admin/packages/xc_admin_frontend/tsconfig.json:19]: for the typechecks yes. It helps the IDEs to identify the type files correctly, especially when you edit the files without compiling them.

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts (hunk 2 above)
Comment (keyvankhademi) [reply governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts:88]: `SetDataSources` implements `PythGovernanceAction` directly so it doesn't inherit from `PythGovernanceActionImpl`.

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts (hunk 4 above)
Comment (keyvankhademi) [reply governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts:55]: 100%. For string keys it's always the same order. Also this is just for the sake of better UI, it really doesn't matter if the order is nor correct.

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/ProposalRow.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            .slice(2)
+            .reduce((acc, curr) => acc + curr, 0)
+          const updatedSummary = [
+            ...shortSummary,
+            ...(otherValue > 0
+              ? ([['other', otherValue]] as [string, number][])
+              : []),
+          ]
+
+          if (!isCancelled) {
```
[the remark this answers is not in the document]
Comment (keyvankhademi) [reply governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/ProposalRow.tsx:103]: When the component dismounts, the state variables become invalid. So if a request is fetching and during that period the component dismounts, without this check we'll get an error that `setState` is not defined.

**Pull request merged**